> I don't know why The test is failed on github but was passed in my local test. 

From details of [check_code_quality](https://circleci.com/gh/huggingface/transformers/1066145):
tests/models/mixtral/test_modeling_mixtral.py:19:8: F401 [*] `random` imported but unused

Unused import of the random module in the test file.

--- a/src/transformers/models/mixtral/modeling_mixtral.py
+++ b/src/transformers/models/mixtral/modeling_mixtral.py
@@ -75,7 +75,7 @@ _CONFIG_FOR_DOC = "MixtralConfig"
 
 
 def load_balancing_loss_func(
-    gate_logits: torch.Tensor, attention_mask: Optional[torch.Tensor], num_experts: torch.Tensor = None, top_k=2
+    gate_logits: torch.Tensor, num_experts: torch.Tensor = None, top_k=2, attention_mask: Optional[torch.Tensor] = None
 ) -> float:
     r"""
     Computes auxiliary load balancing loss as in Switch Transformer - implemented in Pytorch.
@@ -1383,9 +1383,9 @@ class MixtralForCausalLM(MixtralPreTrainedModel):
         if output_router_logits:
             aux_loss = load_balancing_loss_func(
                 outputs.router_logits if return_dict else outputs[-1],
-                attention_mask,
                 self.num_experts,
                 self.num_experts_per_tok,
+                attention_mask,
             )
             if labels is not None:
                 loss += self.router_aux_loss_coef * aux_loss.to(loss.device)  # make sure to reside in the same device

--- a/tests/models/mixtral/test_modeling_mixtral.py
+++ b/tests/models/mixtral/test_modeling_mixtral.py
@@ -15,8 +15,6 @@
 """ Testing suite for the PyTorch Mixtral model. """
 
 
-import math
-import random
 import tempfile
 import unittest
 
@@ -493,8 +491,8 @@ class MixtralModelTest(ModelTesterMixin, GenerationTesterMixin, PipelineTesterMi
         include_padding_result = model(padded_input_ids, attention_mask=None)
 
         # This is to mimic torch.testing.assert_not_close
-        loss_diff = math.fabs(include_padding_result.aux_loss.item() - result.aux_loss.item())
-        self.assertTrue(loss_diff > 1e-2 + 1e-2 * result.aux_loss.item())
+        print(f"loss1={include_padding_result.aux_loss.item()}, loss2={result.aux_loss.item()}")
+        self.assertNotAlmostEqual(include_padding_result.aux_loss.item(), result.aux_loss.item())
 
 
 @require_torch

--- a/tests/models/mixtral/test_modeling_mixtral.py
+++ b/tests/models/mixtral/test_modeling_mixtral.py
@@ -477,9 +477,9 @@ class MixtralModelTest(ModelTesterMixin, GenerationTesterMixin, PipelineTesterMi
 
         # First, we make sure that adding padding tokens doesn't change the loss
         # loss(input_ids, attention_mask=None) == loss(input_ids + padding, attention_mask=attention_mask_with_padding)
-        pad_leng = 1000
+        pad_length = 1000
         # Add padding tokens (assume that pad_token_id=1) to input_ids
-        padding_block = torch.ones(input_ids.shape[0], pad_leng, dtype=torch.int32).to(torch_device)
+        padding_block = torch.ones(input_ids.shape[0], pad_length, dtype=torch.int32).to(torch_device)
         padded_input_ids = torch.cat((padding_block, input_ids), dim=1)  # this is to simulate padding to the left
         padded_attention_mask = padded_input_ids.ne(1).to(torch_device)
 

--- a/tests/models/mixtral/test_modeling_mixtral.py
+++ b/tests/models/mixtral/test_modeling_mixtral.py
@@ -491,7 +491,6 @@ class MixtralModelTest(ModelTesterMixin, GenerationTesterMixin, PipelineTesterMi
         include_padding_result = model(padded_input_ids, attention_mask=None)
 
         # This is to mimic torch.testing.assert_not_close
-        print(f"loss1={include_padding_result.aux_loss.item()}, loss2={result.aux_loss.item()}")
         self.assertNotAlmostEqual(include_padding_result.aux_loss.item(), result.aux_loss.item())
 
 

